tests/utils/test_client.py

```diff
@@ -53,4 +58,38 @@ def test_client_enforce_healthy():
     assert my_client.port == int(CONFIG["my_instance_name"]["port"])
     assert my_client.protocol == CONFIG["my_instance_name"]["protocol"]
     assert my_client.base_path == "/api/versioned/v1/"
-    assert my_client.auth == UsernamePasswordAuth("admin", os.environ["TAMR_TOOLBOX_PASSWORD"],)
+    assert my_client.auth == UsernamePasswordAuth(
+        "admin", os.environ["TAMR_TOOLBOX_PASSWORD"],
+    )
+
+
+@mock_api()
+def test_get_with_connection_retry():
+    # To ensure timeout error is raised to test correct ConnectionError is caught
+
+    log_prefix = "caught_connection_error"
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        log_file_path = os.path.join(
+            tempdir, f"{log_prefix}_{utils.logger._get_log_filename()}"
+        )
+
+    # clears old dir
+    if os.path.exists(log_file_path):
+        os.remove(log_file_path)
+    if os.path.exists(log_file_path):
+        os.remove(log_file_path)
+
+    my_client = utils.client.create(**CONFIG["my_other_instance"])
+    utils.logger.enable_toolbox_logging(
+        log_to_terminal=False, log_directory=log_file_path
```

Comment:
The reason you're getting an `IsDirectory` error is that this needs the directory where the log should be written, instead of the log filepath itself.
```suggestion
        log_to_terminal=False, log_directory=tempdir
```